src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -447,6 +456,23 @@ public boolean isCredentialValid() throws IOException {
     }
 
     /**
+     * Ensures that the API URL is valid.
+     */
+    public boolean isApiUrlValid() throws IOException {
+        try {
+            GHApiInfo apiInfo = retrieve().to("/", GHApiInfo.class);
+            try {
+                new URL(apiInfo.getRateLimitUrl());
```

Comment:
method called `isApiUrlValid()` if it fetches and deserialized -> then true. 
If rate_limit valid -> other logic that can be checked even now.